internal/engine/pod.go

```diff
@@ -122,16 +123,15 @@ func trackPod(ms *store.ManifestState, action k8swatch.PodChangeAction) (*store.
 		(isAncestorMatch && runtime.PodAncestorUID != ancestorUID) {
 		runtime.PodAncestorUID = ancestorUID
 		runtime.Pods = make(map[k8s.PodID]*store.Pod)
-		pod := &store.Pod{
+		podInfo := &store.Pod{
 			PodID:      podID,
 			StartedAt:  startedAt,
 			Status:     status,
 			Namespace:  ns,
 			HasSynclet: hasSynclet,
 		}
-		runtime.Pods[podID] = pod
-		ms.RuntimeState = runtime
```

Comment:
I think this would be clearer if you keep this line in this `if` block and move it immediately after the `runtime.Pods`. The confusion we had the other day about why we're reassigning back to ms.RuntimeState seems like it'd be exacerbated by moving it further away, into `maybeAttachPodToManifest`